internal/resources/attributes.go

```diff
@@ -28,13 +29,75 @@ const (
 	deprecationMessage = "Remove this attribute's configuration as it no longer in use and the attribute will be removed in the next major version of the provider."
 )
 
+type AttributeFunc[T any] func(T)
+
+// withMarkdown sets the MarkdownDescription for the provided attribute.
+// It accepts an attribute of type schema.StringAttribute, schema.Int64Attribute,
+// schema.BoolAttribute, schema.SetAttribute, schema.Float64Attribute,
+// schema.NumberAttribute, or schema.ListAttribute, and a description string.
+// The function returns an AttributeFunc that modifies the attribute.
+func withMarkdown[T interface {
+	*schema.StringAttribute | *schema.Int64Attribute | *schema.BoolAttribute | *schema.SetAttribute |
+	*schema.Float64Attribute | *schema.NumberAttribute | *schema.ListAttribute
+}](description string) AttributeFunc[T] {
+	return func(s T) {
+		switch v := any(s).(type) {
```

Comment:
no point in using generics if you have to do type switch

why not use interface ?

Reply:
looking at this further, with go generics we can't access struct fields.  see https://github.com/golang/go/issues/48522.

so interface is better fit.